packages/core/src/operators/matchPath.operator.ts

```diff
@@ -23,9 +23,37 @@ const pathFactory = (matchers: string[], path: string, suffix?: string): string
 const urlFactory = (path: string): string =>
   removeQueryParams(path);
 
+const paramFactory = (params: RegExpMatchArray | null, routes: pathToRegexp.Token[]): HttpParameters => {
+  if (!params || !routes) { return {}; }
+
+  return routes
+    .filter(route => route.hasOwnProperty('name'))
+    .reduce((obj, route, index) => ({
+      ...obj,
+      [route['name']]: params[++index],
+    }), {});
+};
+
+const routeFactory = (req: HttpRequest, path: string, opts: MatcherOpts = {}): HttpRoute => {
+  const match = pathFactory(req.matchers!, path, opts.suffix);
+  const url = urlFactory(req.url!);
+  const params = pathToRegexp(match).exec(url);
+  const routes = pathToRegexp.parse(match);
+
+  return {
+    url,
+    params: paramFactory(params, routes),
+  };
+};
+
 export const matchPath = (path: string, opts: MatcherOpts = {}) => (source$: Observable<HttpRequest>) =>
   source$.pipe(
-    tap(req => req.matchers = req.matchers || []),
+    tap(req => {
+      req.matchers = req.matchers || [];
+      req.route = {
```

Comment:
Move it after `filter` operator. Current implementation will update `req.route` for each endpoint - but it should be applied only for matched (filtered) ones.